### PR TITLE
add Vietnamese

### PR DESCRIPTION
**Pull request by vdhan**

### CODE REVIEW

**Review by IvanSavenko — APPROVED** [review]

**Comment by Laserlicht** [comment]
If you like, you can also translate:
- Launcher: https://github.com/vcmi/vcmi/blob/develop/launcher/translation/english.ts
- Mapeditor: https://github.com/vcmi/vcmi/blob/develop/mapeditor/translation/english.ts

With this tool: https://github.com/lelegard/qtlinguist-installers/releases/tag/v5.12.0

And Android:
- https://github.com/vcmi/vcmi/blob/develop/android/vcmi-app/src/main/res/values/strings.xml

Especially the Launcher would be great. It's the first thing that would a player see after starting VCMI.

**Comment by vdhan** [comment]
> If you like, you can also translate:
> 
>     * Launcher: https://github.com/vcmi/vcmi/blob/develop/launcher/translation/english.ts
> 
>     * Mapeditor: https://github.com/vcmi/vcmi/blob/develop/mapeditor/translation/english.ts
> 
> 
> With this tool: https://github.com/lelegard/qtlinguist-installers/releases/tag/v5.12.0
> 
> And Android:
> 
>     * https://github.com/vcmi/vcmi/blob/develop/android/vcmi-app/src/main/res/values/strings.xml
> 
> 
> Especially the Launcher would be great. It's the first thing that would a player see after starting VCMI.

Thank you for your comment. I translated Launcher and Mapeditor already, but don't know how to apply it to vcmi

**Comment by IvanSavenko** [comment]
> Thank you for your comment. I translated Launcher and Mapeditor already, but don't know how to apply it to vcmi

If you have translated .ts files then:
- Place it at the same location as other translations, e.g. (VCMI sources)/launcher/translation/vietnamese.ts
- Open (VCMI sources)/launcher/CMakeLists.txt with text editor
- add new translation to the list at line 60 (Launcher) or line 100 (Map Editor)
- Commit and push using git, like you did with json's

If you want to test translation you can select in Qt Linguist "Export as", navigate to `(VCMI install)/launcher/translation` and save exported file in this directory. VCMI should be able to use this file when you select Vietnamese in Launcher.

**Comment by vdhan** [comment]
> > Thank you for your comment. I translated Launcher and Mapeditor already, but don't know how to apply it to vcmi
> 
> If you have translated .ts files then:
> 
>     * Place it at the same location as other translations, e.g. (VCMI sources)/launcher/translation/vietnamese.ts
> 
>     * Open (VCMI sources)/launcher/CMakeLists.txt with text editor
> 
>     * add new translation to the list at line 60 (Launcher) or line 100 (Map Editor)
> 
>     * Commit and push using git, like you did with json's
> 
> 
> If you want to test translation you can select in Qt Linguist "Export as", navigate to `(VCMI install)/launcher/translation` and save exported file in this directory. VCMI should be able to use this file when you select Vietnamese in Launcher.

Thank you. Let me try

File: Mods/vcmi/mod.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	"chinese" : {
 		"name" : "VCMI essential files",
 		"description" : "Essential files required for VCMI to run correctly",
-
```
Comment (Nordsoft91) [inline Mods/vcmi/mod.json:8]: understood this is auto-formatting, but maybe let's exclude them from PR

Comment (vdhan) [reply Mods/vcmi/mod.json:8]: It just removes trailing spaces, I can convert it. Could you please tell me reason why we keep trailing spaces on blank lines?

Comment (nullkiller) [reply Mods/vcmi/mod.json:8]: I think the idea was to not apply any autoformatting on things which do not relate directly to your PR. While these changes are small and it is ease to review them in general such approach can cause a lot of extra changes.

Comment (nullkiller) [reply Mods/vcmi/mod.json:8]: Still I do not see any issues with removing spaces so think we can accept it as is.

**Comment by IvanSavenko** [comment]
Seems to be working

**Pull request merged**